manager/scheduler/scheduler.go

```diff
@@ -286,17 +287,63 @@ func (s *Scheduler) deleteTask(ctx context.Context, t *api.Task) {
 	}
 }
 
+func reconcileGenericResources(nodeRes *api.Resources, assigned map[string]*api.GenericResource) {
+	for taskRes, taskVal := range assigned {
+		nodeVal := nodeRes.Generic[taskRes]
+		reconcileGenericResource(nodeVal, taskVal)
+	}
+}
+
+func reconcileGenericResource(nodeVal, taskVal *api.GenericResource) {
+
```

Comment:
blank line here

I think `nodeVal` can end up being nil here if the node was updated to no longer have that type of resource. Either we should check for this here, or above in `reconcileGenericResource`.